test/integration/test-amp-inabox-rendering.js

```diff
@@ -0,0 +1,165 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  RequestBank,
+} from '../../testing/test-helper';
+
+// The image ad as seen in examples/inabox.gpt.html,
+// with visibility pings being placeholders that's substituted with calls to the
+// request bank.
+const adBody = __html__['test/fixtures/amp-cupcake-ad.html'] // eslint-disable-line no-undef
+    .replace(/__VIEW_URL__/g, RequestBank.getUrl('view')) // get all instances
+    .replace('__VISIBLE_URL__', RequestBank.getUrl('visible'))
+    .replace('__ACTIVE_VIEW_URL__', RequestBank.getUrl('activeview'));
+
+function testVisibilityPings(visibleDelay, activeViewDelay) {
+  let viewTime = 0;
+  return RequestBank.withdraw('view').then(() => {
```

Comment:
I felt the following lines can be a bit cleaner than chaining all promises.
```
const viewPromise = RequestBank.withdraw('view').then(() => viewTime = Date.now());
const visiblePromise = RequestBank.withdraw('visible').then(() => visibleTime = Date.now());

Promise.all([viewPromise, visiblePromise]).then...
```

Reply:
Done. Thanks for the help!